hazelcast/include/hazelcast/client/serialization/pimpl/compact.h

```diff
@@ -0,0 +1,426 @@
+     * in the schema.
+     */
+    int32_t read_int32(const std::string& field_name);
+
+    /**
+     * Reads a 32-bit two's complement signed integer or returns the default
+     * value.
+     *
+     * @param fieldName    name of the field.
+     * @param defaultValue default value to return if the field with the given
+     * name does not exist in the schema or the type of the field does not match
+     * with the one defined in the schema.
+     * @return the value or the default value of the field.
+     */
+    int32_t read_int32(const std::string& field_name, int32_t default_value);
+
+    /**
+     * Reads an UTF-8 encoded string.
+     *
+     * @param fieldName name of the field.
+     * @return the value of the field.
+     * @throws hazelcast_serialization if the field does not exist in
+     * the schema or the type of the field does not match with the one defined
+     * in the schema.
+     */
+    boost::optional<std::string> read_string(const std::string& field_name);
+
+    /**
+     * Reads an UTF-8 encoded string or returns the default value.
+     *
+     * @param fieldName    name of the field.
+     * @param defaultValue default value to return if the field with the given
+     * name does not exist in the schema or the type of the field does not match
+     * with the one defined in the schema.
+     * @return the value or the default value of the field.
+     */
+    boost::optional<std::string> read_string(
+      const std::string& field_name,
+      const boost::optional<std::string>& default_value);
+
+private:
+    compact_reader(pimpl::compact_stream_serializer& compact_stream_serializer,
+                   object_data_input& object_data_input,
+                   const pimpl::schema& schema);
+
+    friend compact_reader pimpl::create_compact_reader(
+      pimpl::compact_stream_serializer& compact_stream_serializer,
+      object_data_input& object_data_input,
+      const pimpl::schema& schema);
+
+    bool is_field_exists(const std::string& fieldName,
+                         enum pimpl::field_kind kind);
+    const pimpl::field_descriptor& get_field_descriptor(
+      const std::string& field_name) const;
+    const pimpl::field_descriptor& get_field_descriptor(
+      const std::string& field_name,
+      enum pimpl::field_kind field_kind) const;
+    exception::hazelcast_serialization unknown_field_exception(
+      const std::string& field_name) const;
+    exception::hazelcast_serialization unexpected_field_kind(
+      enum pimpl::field_kind field_kind,
+      const std::string& field_name) const;
+    template<typename T>
+    boost::optional<T> get_variable_size(
+      const pimpl::field_descriptor& field_descriptor);
+    template<typename T>
+    boost::optional<T> get_variable_size(const std::string& field_name,
+                                         enum pimpl::field_kind field_kind);
+    template<typename T>
+    T get_variable_as_non_null(const pimpl::field_descriptor& field_descriptor,
```

Comment:
Fixed here 7d6214369fe1cd0b922d38d78739ef4d97303582

Reply:
Since we are following java, and same names make it easier to match both sides. If we are changing the names(this and others) different from Java, can we also change it in the same way for Java (let's at least open the issue to change Java as well)?

Reply:
It is already `get_variable_size_as_non_null` on java. It was a mistake that I did in the first place.